### PR TITLE
util/execdetails: remove pingcap/check for execdetails_test

### LINKED ISSUE #26285: migrate test-infra to testify for `util/execdetails` pkg

### PR DESCRIPTION
**Pull request by jyz0309**
### What problem does this PR solve?
Issue Number: close #26285 

Problem Summary:
util/execdetails: remove pingcap/check for execdetails_test

What's Changed:
util/execdetails: remove pingcap/check and use testify for execdetails_test

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note 

- No Release note

### CODE REVIEW

**Review by xhebox — COMMENTED** [review]
Looks like tests did not make use of checkers at all? @tisonkun What do you think, would `require` and `assert` be better?

**Review by tisonkun — CHANGES_REQUESTED** [review]
@xhebox generally `require` acts the same as `assert` while calls `FailNow` if assertion fails.

@jyz0309 I'd suggest you make use of testify matchers to simplify the code, for example,

```go
if stats.String() != expect {
		t.Fatalf("%v != %v", stats.String(), expect)
	}
```

can be 

```go
require.Equal(t, expect, stats.String())
```

**Comment by jyz0309** [comment]
> @xhebox generally `require` acts the same as `assert` while calls `FailNow` if assertion fails.
> 
> @jyz0309 I'd suggest you make use of testify matchers to simplify the code, for example,
> 
> ```go
> if stats.String() != expect {
> 		t.Fatalf("%v != %v", stats.String(), expect)
> 	}
> ```
> 
> can be
> 
> ```go
> require.Equal(t, expect, stats.String())
> ```

OK，Will finish it this weekend

File: util/execdetails/execdetails_test.go [hunk 1]
```diff
@@ -111,51 +105,39 @@ func TestCopRuntimeStats(t *testing.T) {
...
-	}
+	expected := "tikv_task:{proc max:2ns, min:1ns, p80:2ns, p95:2ns, iters:3, tasks:2}, " +
+		"scan_detail: {total_process_keys: 10, total_keys: 15, rocksdb: {delete_skipped_count: 5, key_skipped_count: 1, block: {cache_hit_count: 10, read_count: 20, read_byte: 100 Bytes}}}"
+	assert.Equal(t, expected, cop.String())
+
 	copStats := cop.stats["8.8.8.8"]
-	if copStats == nil {
-		t.Fatal("cop stats is nil")
-	}
+	assert.NotNil(t, copStats)
```
Comment (xhebox) [inline util/execdetails/execdetails_test.go:116]: `require.NotNil` will fail faster and better.

**Review by xhebox — COMMENTED** [review]
`TestFormatDurationForExplain`, `TestRootRuntimeStats`, `TestRuntimeStatsWithCommit`, `TestCopRuntimeStatsForTiFlash` may make use of `assert/require`, too.

**Comment by jyz0309** [comment]
cc @tisonkun

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by xhebox — APPROVED** [review]

**Comment by tisonkun** [comment]
/cc @mmyj @tiancaiamao 

May you help on reviewing this PR? Thanks!

**Review by tiancaiamao — APPROVED** [review]

**Review by lysu — APPROVED** [review]

**Pull request merged**